packages/template-retail-react-app/app/commerce-api/hooks/useBasket.js

```diff
@@ -4,418 +4,420 @@
-            },
-
-            /**
-             * Fetches and returns promo details for the given IDs
-             * @param {Array<string>} ids - The promo ids to fetch
-             * @returns {Object} - API response containing data
-             */
-            async getPromotions(ids) {
-                const response = await api.shopperPromotions.getPromotions({
-                    parameters: {ids: ids.join(',')}
-                })
-
-                return response
-            },
-
-            /**
-             * Creates an order using the current basket.
-             */
-            async createOrder() {
-                const response = await api.shopperOrders.createOrder({
-                    body: {basketId: basket.basketId}
-                })
-
-                if (response.fault || (response.title && response.type && response.detail)) {
-                    throw new Error(response.title)
-                }
-
-                // We replace the basket with the order result data so we can display
-                // it on the confirmation page. The basket is automatically deleted
-                // in SF so we need to make sure a new one is created when leaving the confirmation.
-                setBasket(response)
-            },
-
-            /**
-             * Fetches the applicable shipping methods for the current basket
-             * @returns {Object} - API response containing data
-             */
-            getShippingMethods() {
-                return api.shopperBaskets.getShippingMethodsForShipment({
-                    parameters: {basketId: basket.basketId, shipmentId: 'me'}
-                })
-            },
-
-            /**
-             * Merge data from the previous shopper's basket into the current shopper's active basket
-             * and delete the previous shopper's basket.
-             */
-            async mergeBasket() {
-                const response = api.shopperBaskets.mergeBasket({
-                    headers: {
-                        'Content-Type': 'application/json' // This is not required since the request has no body but CommerceAPI throws a '419 - Unsupported Media Type' error if this header is removed.
-                    },
-                    parameters: {
-                        createDestinationBasket: true // If the current shopper has an active basket, this parameter is ignored.
-                    }
-                })
-
-                if (response.fault) {
-                    throw new Error(response)
-                }
-
-                setBasket(response)
-            }
-        }
-    }, [customer, basket, setBasket])
-
-    return self
-}
+// import {useContext, useMemo} from 'react'
+// import useEinstein from './useEinstein'
+// import {useCommerceAPI, BasketContext} from '../contexts'
+// import useCustomer from './useCustomer'
+// import {isError} from '../utils'
+
+export {default} from 'pwa-kit-commerce/commerce-api/hooks/useBasket'
```

Comment:
Since I moved the commerce-api hooks logic into a different package, and to avoid so many code change inside the Retail App, I simply re-export the hook in here. The real implementation should not look like this. The hooks should be imported from the library where it is being used.